src/ffishjs.cpp

```diff
@@ -263,6 +263,34 @@ class Board {
     return pos.game_ply();
   }
 
+  int game_result() const {
+    Value result;
```

Comment:
The idea of using `Value` here looks good.
However, I'm unsure if this is really meaningful to the user.
For instance `VALUE_KNOWN_WIN` corresponds to `10000` and `VALUE_MATE` corresponds to `32000`.
There is also the option to make enums available in javascript, similar to:
```C++
enum_<Notation>("Notation")
```
Generally, I'd like to avoid introducing to many enums as there are not default types in JavaScript.
Perhaps, it's best to keep this as a simple int.
In this case, we should give a detailed comment in `Fairy-Stockfish/tests/js/test.js` what this int values mean.

Reply:
In python-chess there is method called `outcome(*, claim_draw: bool = False) ` and an `enum` called `class chess.Termination(value)`.
It could make sense to use these as well here.

Reply:
These are great comments - thank you!

For the `Value`, I was simply copying the way pyffish did it. But I like the way python-chess does it a lot better. I have a few thoughts:

If we wanted an enum, we could create a function `board.outcome()` with the values: `CHECKMATE`, `STALEMATE`, `INSUFFICIENT_MATERIAL`, `N_MOVE_RULE`, `N_FOLD_RULE`, and `VARIANT_END_CONDITION` or something along those lines.

In python-chess, the `chess.Outcome` object has a function `result()` which is a string, so I was also wondering if we wanted to have a function `board.result()` which returns a string in "1-0", "0-1", and "1/2-1/2". I feel like creating a whole `Outcome` object for ffish.js is probably overkill. Though maybe not, I'd love to hear your thoughts!

Reply:
On this topic - currently the function `board.isGameOver()` only checks for a lack of legal moves. Should this function be adjusted to also account for conditions such as insufficient material/optional draws?